src/JSInterop/Microsoft.JSInterop/test/Infrastructure/DotNetDispatcherTest.cs

```diff
@@ -479,15 +479,14 @@ public async Task CanInvokeAsyncMethod()
             // Assert: Correct completion information
             Assert.Equal(callId, jsRuntime.LastCompletionCallId);
             Assert.True(jsRuntime.LastCompletionResult.Success);
-            var result = Assert.IsType<object[]>(jsRuntime.LastCompletionResult.Result);
-            var resultDto1 = Assert.IsType<TestDTO>(result[0]);
+            var resultJson = Assert.IsType<string>(jsRuntime.LastCompletionResult.ResultJson);
+            var result = JsonSerializer.Deserialize<SomePublicType.InvokableAsyncMethodResult>(resultJson, jsRuntime.JsonSerializerOptions);
```

Comment:
This test was a bit weird before because it looked as if it was testing how the results get marshalled and represented as JSON, but it never actually JSON serialized anything and was cheating by looking at the unserialized results directly.

I've changed it now (in fact, was forced to by the clearer layering) so now it actually does observe things about the marshalling. As part of this I had to change it from a hetrogeneous array to some new DTO so that S.T.J. could deserialize it.